src/transcript/TranscriptEvent.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import DataMessage from '../datamessage/DataMessage';
+import {
+  SdkTranscriptFrame,
+  SdkTranscriptionStatus,
+  SdkTranscriptItem,
+} from '../signalingprotocol/SignalingProtocol';
+import Transcript from './Transcript';
+import TranscriptAlternative from './TranscriptAlternative';
+import TranscriptionStatus from './TranscriptionStatus';
+import TranscriptionStatusType from './TranscriptionStatusType';
+import TranscriptItem from './TranscriptItem';
+import TranscriptItemType from './TranscriptItemType';
+import TranscriptResult from './TranscriptResult';
+
+const TranscriptionStatusTypes = {
+  [SdkTranscriptionStatus.Type.STARTED]: TranscriptionStatusType.STARTED,
+  [SdkTranscriptionStatus.Type.INTERRUPTED]: TranscriptionStatusType.INTERRUPTED,
+  [SdkTranscriptionStatus.Type.RESUMED]: TranscriptionStatusType.RESUMED,
+  [SdkTranscriptionStatus.Type.STOPPED]: TranscriptionStatusType.STOPPED,
+  [SdkTranscriptionStatus.Type.FAILED]: TranscriptionStatusType.FAILED,
+};
+
+export default class TranscriptEvent {
+  // An TranscriptEvent can contain one of the following,
+  // either the transcription status, or transcript
+  status?: TranscriptionStatus;
+  transcript?: Transcript;
```

Comment:
The way this is normally modeled in TypeScript, by the way, is as a union:

```typescript
type TranscriptEvent = TranscriptionStatus | Transcript;
```

or as a [discriminated union](https://thoughtbot.com/blog/the-case-for-discriminated-union-types-with-typescript):

https://github.com/aws/amazon-chime-sdk-js/blob/cbaada44518a6bc2b531b05d3d2a961e77ca9d6c/libs/voicefocus/voicefocus.d.ts#L24-L31

— what you've written here allows a `TranscriptEvent` with neither a status nor a transcript, which is invalid. Good type systems can make invalid states unrepresentable.

If Kyu or others have comments to address, consider tidying this up.

Reply:
Updated to make `TranscriptEvent` a union type instead of a class.